### PR TITLE
Rspec test for Reservations

### LINKED ISSUE #6: [1pt] Rspec for Reservation (Ranjit)
- Unit test for reservation

### PR DESCRIPTION
**Pull request by aboongm**
- [x]  Rspec test for reservation model
- [x]  Update the factories with reservation creation data
- [x]  Rspec test for reservation requests

### CODE REVIEW

File: spec/models/reservation_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+RSpec.describe Reservation, type: :model do
+  before(:all) do
+    @user = User.create(username: "john doe1#{SecureRandom.hex}", email: "jod#{SecureRandom.hex}@test.com", password: '123456')
+    expect(@user).to be_valid
+    @doctor = Doctor.create!(name: 'doc1', detail: 'doctor details1', photo: 'some url1', city: 'some city1', specialization: 'specialization1', fee: 100.00)
+    @reservation = Reservation.new(city: 'Imphal', date: '23/8/2022', user_id: @user.id, doctor_id: @doctor.id)
+  end
```
Comment (steveWDamesJr) [inline spec/models/reservation_spec.rb:9]: Great job with this preparation of users, doctor, and reservation as setup to perform your tests! 💯

File: spec/models/reservation_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      expect(@reservation).to_not be_valid
+    end
+    it 'should not have date that is blank' do
+      @reservation.date = ''
+      expect(@reservation).to_not be_valid
+    end
+    it 'should not have date that is more than 10 characters' do
+      @reservation.date = '23/8/2022' * 2
+      expect(@reservation).to_not be_valid
+    end
```
Comment (steveWDamesJr) [inline spec/models/reservation_spec.rb:45]: Great job implementing reservation model tests! 🎉

File: spec/requests/reservations_spec.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+      expect(response).to have_http_status(200)
+    end
+  end
+
+  describe 'DELETE #destroy' do
+    it 'destroys the requested reservation' do
+      reservation = FactoryBot.create(:reservation, city: 'Imphal', date: '24/08/2022', user_id: @current_user.id, doctor_id: 1)
+      expect do
+        delete :destroy, params: { id: reservation.id }
```
Comment (steveWDamesJr) [inline spec/requests/reservations_spec.rb:48]: All requests spec tests for reservations passing 🙌

**Review by steveWDamesJr — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @aboongm, 
✅ Great uses of testing suite to perform tests
✅ All tests pass

Your project is complete! Keep up the awesomeness!!!

There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/reservations_spec.rb (hunk 3 above)
Comment (aboongm) [reply spec/requests/reservations_spec.rb:48]: Thanks, Steve!

**Pull request merged**